machine/machine_core/ssh_connection.py

```diff
@@ -69,6 +69,22 @@ def message(self, *args):
         if self.verbose:
             sys.stderr.write(" ".join(args) + '\n')
 
+    def print_console_log(self):
+        """Prints VM's console to stderr"""
+        console_file = "/tmp/console.log"
+        try:
+            f = open(console_file)
+            console_log = f.read().strip()
```

Comment:
Please reduce the scope of the try/except to the open/read pair, and catch all `OSError`s -- this is already in an exception handler, so this better not crash.  Print the exception when it happens for debugging.

Please also use a `with` context (see codeql)

Reply:
Fixed